insights/core/hydration.py

```diff
@@ -8,13 +8,21 @@
 
 log = logging.getLogger(__name__)
 
-
-def get_all_files(path):
-    all_files = []
-    for f in archives.get_all_files(path):
-        if os.path.isfile(f) and not os.path.islink(f):
-            all_files.append(f)
-    return all_files
+if hasattr(os, "scandir"):
+    def get_all_files(path):
+        with os.scandir(path) as it:
+            for ent in it:
+                if ent.is_dir(follow_symlinks=False):
+                    for pth in get_all_files(ent.path):
+                        yield pth
+                elif ent.is_file(follow_symlinks=False):
+                    yield ent.path
+
+else:
+    def get_all_files(path):
```

Comment:
I think we should remove `get_all_files` from insights.core.archives and just have the alternate implementations defined here since it's not used anywhere else.

Reply:
Like so?